### ISSUE TITLE
Go-to-definition for use of C++20 concept

### ISSUE THREAD

**Issue by HighCommander4** [clangd/clangd#260]
In the following code example:

```c++
template <typename T>
concept [[Fooable]] = requires (T t) { t.foo(); };

template <typename T> requires F^ooable<T>
void bar(T t) {
  t.foo();
}
```

go-to-definition at the indicated location (which is a use of a C++20 concept name) should navigate to the indicated definition of the concept.

**Comment by HighCommander4** [comment]
[Proposed fix](https://reviews.llvm.org/D73140)

**Comment by sam-mccall** [comment]
This was fixed in llvm/llvm-project@cbcd07a4815f546739fb1374a7f24f35e5352e17
